### ISSUE TITLE
Invalid text representation: 7

### ISSUE THREAD

**Issue by ownhopper** [owncloud/core#27960]
### Expected behaviour
Smooth uploading file without error.

### Actual behaviour
Uploading a file via web interface I get a small yellow pop up on the top saying "Internal Server Error". Refreshing the page, file have been saved.

### Server configuration
**Operating system**:

Debian Jessie 8.8 - 3.16.0-4-amd64 #1 SMP Debian 3.16.43-2 (2017-04-30) x86_64 GNU/Linux

**Web server:**

Apache 2.4.10

**Database:**

PostgreSQL 9.4.12

**PHP version:**

PHP 5.6.30-0+deb8u1

**ownCloud version:** (see ownCloud admin page)

10.0.0.12

**Updated from an older ownCloud or fresh install:**

Fresh install.

**Where did you install ownCloud from:**

Archive file.

**Signing status (ownCloud 9.0 and above):**

```
No errors have been found.
```

**The content of config/config.php:**

```
{
    "system": {
        "instanceid": "oc8ixrmtnw6y",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "***REMOVED SENSITIVE VALUE***"
        ],
        "datadirectory": "\/var\/www\/my_domain\/data",
        "overwrite.cli.url": "https:\/\/my_domain\/owncloud",
        "dbtype": "pgsql",
        "version": "10.0.0.12",
        "dbname": "owncloud",
        "dbhost": "localhost:5432",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "logtimezone": "Europe\/Rome",
        "installed": true,
        "mail_smtpmode": "smtp",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_domain": "gmail.com",
        "mail_smtpauthtype": "LOGIN",
        "mail_smtpauth": 1,
        "mail_smtphost": "smtp.gmail.com",
        "mail_smtpport": "587",
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpsecure": "tls"
    }
}
```

**List of activated apps:**

```
Enabled:
  - activity: 2.3.3
  - comments: 0.3.0
  - configreport: 0.1.1
  - dav: 0.2.8
  - federatedfilesharing: 0.3.0
  - federation: 0.1.0
  - files: 1.5.1
  - files_antivirus: 0.10.0.0
  - files_external: 0.7.0
  - files_pdfviewer: 0.8.1
  - files_sharing: 0.10.0
  - files_texteditor: 2.1
  - files_trashbin: 0.9.0
  - files_versions: 1.3.0
  - files_videoplayer: 0.9.8
  - firstrunwizard: 1.1
  - gallery: 15.0.0
  - market: 0.1.0
  - notifications: 0.3.0
  - provisioning_api: 0.5.0
  - systemtags: 0.3.0
  - templateeditor: 0.1
  - updatenotification: 0.2.1
Disabled:
  - encryption
  - example-theme
  - external
  - user_external
```

**Are you using external storage, if yes which one:** No

**Are you using encryption:** No

**Are you using an external user-backend, if yes which one:** No

### Client configuration
**Browser:**

Chromium 57.0.2987.98 Built on 8.7, running on Debian 8.8 (64-bit)

**Operating system:**

Debian Jessie 8.8

#### Web server error log
```
Nothing logged.
```

#### ownCloud log (data/owncloud.log)
```
{"reqId":"eLy2C6A1vJldLlAtVGUy","level":4,"time":"2017-05-19T00:43:34+02:00","remoteAddr":"ip_address","user":"my_user","app":"webdav","method":"PUT","url":"\owncloud\remote.php\webdav\garbage\2017YY0172253.pdf","message":"Exception: {"Message":"An exception occurred while executing 'UPDATE "oc_filecache" SET "mtime" = GREATEST("mtime", ?), "etag" = ? WHERE ("storage" = ?) AND ("path_hash" IN ('d41d8cd98f00b204e9800998ecf8427e', '45b963397aa40d4a0063e0d85e4fe7a1', '092555b67f6aaf450673eefeec1bd065'))' with params ["1495145306.149", "591e2396b822e", 1]:\\n\\nSQLSTATE[22P02]: Invalid text representation: 7 ERROR: invalid input syntax for integer type: "1495145306.149"","Exception":"Doctrine\DBAL\Exception\DriverException","Code":0,"Trace":"#0 \var\www\owncloud\lib\composer\doctrine\dbal\lib\Doctrine\DBAL\DBALException.php(128): Doctrine\DBAL\Driver\AbstractPostgreSQLDriver->convertException('An exception oc...', Object(Doctrine\DBAL\Driver\PDOException))\\n#1 \var\www\owncloud\lib\composer\doctrine\dbal\lib\Doctrine\DBAL\Connection.php(1015): Doctrine\DBAL\DBALException::driverExceptionDuringQuery(Object(Doctrine\DBAL\Driver\PDOPgSql\Driver), Object(Doctrine\DBAL\Driver\PDOException), 'UPDATE "oc_file...', Array)\\n#2 \var\www\owncloud\lib\private\DB\Connection.php(211): Doctrine\DBAL\Connection->executeUpdate('UPDATE "oc_file...', Array, Array)\\n#3 \var\www\owncloud\lib\composer\doctrine\dbal\lib\Doctrine\DBAL\Query\QueryBuilder.php(208): OC\DB\Connection->executeUpdate('UPDATE `*PREFIX...', Array, Array)\\n#4 \var\www\owncloud\lib\private\DB\QueryBuilder\QueryBuilder.php(141): Doctrine\DBAL\Query\QueryBuilder->execute()\\n#5 \var\www\owncloud\lib\private\Files\Cache\Propagator.php(87): OC\DB\QueryBuilder\QueryBuilder->execute()\\n#6 \var\www\owncloud\lib\private\Files\Cache\HomePropagator.php(48): OC\Files\Cache\Propagator->propagateChange('files\garbage\2...', '1495145306.149', 0)\\n#7 \var\www\owncloud\lib\private\Files\Cache\Updater.php(138): OC\Files\Cache\HomePropagator->propagateChange('files\garbage\2...', '1495145306.149', 0)\\n#8 \var\www\owncloud\lib\private\Files\View.php(314): OC\Files\Cache\Updater->update('files\garbage\2...', '1495145306.149')\\n#9 \var\www\owncloud\lib\private\Files\View.php(1141): OC\Files\View->writeUpdate(Object(OC\Files\Storage\Wrapper\Checksum), 'files\garbage\2...', '1495145306.149')\\n#10 \var\www\owncloud\lib\private\Files\View.php(551): OC\Files\View->basicOperation('touch', '\garbage\2017YY...', Array, '1495145306.149')\\n#11 \var\www\owncloud\apps\dav\lib\Connector\Sabre\File.php(215): OC\Files\View->touch('\garbage\2017YY...', '1495145306.149')\\n#12 \var\www\owncloud\apps\dav\lib\Connector\Sabre\Directory.php(158): OCA\DAV\Connector\Sabre\File->put(Resource id #66)\\n#13 \var\www\owncloud\lib\composer\sabre\dav\lib\DAV\Server.php(1095): OCA\DAV\Connector\Sabre\Directory->createFile('2017YY0172253.p...', Resource id #66)\\n#14 \var\www\owncloud\lib\composer\sabre\dav\lib\DAV\CorePlugin.php(525): Sabre\DAV\Server->createFile('garbage\2017YY0...', Resource id #66, NULL)\\n#15 [internal function]: Sabre\DAV\CorePlugin->httpPut(Object(Sabre\HTTP\Request), Object(Sabre\HTTP\Response))\\n#16 \var\www\owncloud\lib\composer\sabre\event\lib\EventEmitterTrait.php(105): call_user_func_array(Array, Array)\\n#17 \var\www\owncloud\lib\composer\sabre\dav\lib\DAV\Server.php(479): Sabre\Event\EventEmitter->emit('method:PUT', Array)\\n#18 \var\www\owncloud\lib\composer\sabre\dav\lib\DAV\Server.php(254): Sabre\DAV\Server->invokeMethod(Object(Sabre\HTTP\Request), Object(Sabre\HTTP\Response))\\n#19 \var\www\owncloud\apps\dav\appinfo\v1\webdav.php(63): Sabre\DAV\Server->exec()\\n#20 \var\www\owncloud\remote.php(165): require_once('\var\www\own...')\\n#21 {main}","File":"\var\www\owncloud\lib\composer\doctrine\dbal\lib\Doctrine\DBAL\Driver\AbstractPostgreSQLDriver.php","Line":91,"User":"my_user"}"}
```

#### Browser log
```
jquery-migrate.min.js:2 JQMIGRATE: Migrate is installed, version 1.4.0
js.js:2285 Deprecation warning: tipsy is deprecated. Use tooltip instead.
jQuery.fn.tipsy @ js.js:2285
(anonymous) @ file-upload.js:1011
dispatch @ jquery.min.js:3
r.handle @ jquery.min.js:3
trigger @ jquery.min.js:3
a.event.trigger @ jquery-migrate.min.js:2
(anonymous) @ jquery.min.js:3
each @ jquery.min.js:2
each @ jquery.min.js:2
trigger @ jquery.min.js:3
_trigger @ jquery-ui.custom.js:787
_beforeSend @ jquery.fileupload.js:758
(anonymous) @ jquery-ui.custom.js:405
_onSend @ jquery.fileupload.js:869
(anonymous) @ jquery-ui.custom.js:405
data.submit @ jquery.fileupload.js:612
(anonymous) @ file-upload.js:260
(anonymous) @ jquery.min.js:2
j @ jquery.min.js:2
add @ jquery.min.js:2
(anonymous) @ jquery.min.js:2
each @ jquery.min.js:2
(anonymous) @ jquery.min.js:2
a.Deferred @ jquery-migrate.min.js:2
then @ jquery.min.js:2
submit @ file-upload.js:259
(anonymous) @ file-upload.js:498
_.each._.forEach @ underscore.js:153
submitUploads @ file-upload.js:496
onNoConflicts @ file-upload.js:888
checkExistingFiles @ file-upload.js:700
add @ file-upload.js:906
_trigger @ jquery-ui.custom.js:789
(anonymous) @ jquery.fileupload.js:935
each @ jquery.min.js:2
_onAdd @ jquery.fileupload.js:928
(anonymous) @ jquery-ui.custom.js:405
(anonymous) @ jquery.fileupload.js:1105
j @ jquery.min.js:2
add @ jquery.min.js:2
always @ jquery.min.js:2
_onChange @ jquery.fileupload.js:1099
(anonymous) @ jquery-ui.custom.js:405
handlerProxy @ jquery-ui.custom.js:706
dispatch @ jquery.min.js:3
r.handle @ jquery.min.js:3
js.js:2285 Deprecation warning: tipsy is deprecated. Use tooltip instead.
jQuery.fn.tipsy @ js.js:2285
start @ file-upload.js:919
_trigger @ jquery-ui.custom.js:789
_beforeSend @ jquery.fileupload.js:758
(anonymous) @ jquery-ui.custom.js:405
_onSend @ jquery.fileupload.js:869
(anonymous) @ jquery-ui.custom.js:405
data.submit @ jquery.fileupload.js:612
(anonymous) @ file-upload.js:260
(anonymous) @ jquery.min.js:2
j @ jquery.min.js:2
add @ jquery.min.js:2
(anonymous) @ jquery.min.js:2
each @ jquery.min.js:2
(anonymous) @ jquery.min.js:2
a.Deferred @ jquery-migrate.min.js:2
then @ jquery.min.js:2
submit @ file-upload.js:259
(anonymous) @ file-upload.js:498
_.each._.forEach @ underscore.js:153
submitUploads @ file-upload.js:496
onNoConflicts @ file-upload.js:888
checkExistingFiles @ file-upload.js:700
add @ file-upload.js:906
_trigger @ jquery-ui.custom.js:789
(anonymous) @ jquery.fileupload.js:935
each @ jquery.min.js:2
_onAdd @ jquery.fileupload.js:928
(anonymous) @ jquery-ui.custom.js:405
(anonymous) @ jquery.fileupload.js:1105
j @ jquery.min.js:2
add @ jquery.min.js:2
always @ jquery.min.js:2
_onChange @ jquery.fileupload.js:1099
(anonymous) @ jquery-ui.custom.js:405
handlerProxy @ jquery-ui.custom.js:706
dispatch @ jquery.min.js:3
r.handle @ jquery.min.js:3
jquery.min.js:4 PUT https://my_domain/owncloud/remote.php/webdav/garbage/2017YY0172253.pdf 500 (Internal Server Error)
send @ jquery.min.js:4
ajax @ jquery.min.js:4
send @ jquery.fileupload.js:834
(anonymous) @ jquery-migrate.min.js:2
j @ jquery.min.js:2
add @ jquery.min.js:2
(anonymous) @ jquery-migrate.min.js:2
each @ jquery.min.js:2
(anonymous) @ jquery-migrate.min.js:2
a.Deferred @ jquery-migrate.min.js:2
c.pipe.e.pipe @ jquery-migrate.min.js:2
_onSend @ jquery.fileupload.js:878
(anonymous) @ jquery-ui.custom.js:405
data.submit @ jquery.fileupload.js:612
(anonymous) @ file-upload.js:260
(anonymous) @ jquery.min.js:2
j @ jquery.min.js:2
add @ jquery.min.js:2
(anonymous) @ jquery.min.js:2
each @ jquery.min.js:2
(anonymous) @ jquery.min.js:2
a.Deferred @ jquery-migrate.min.js:2
then @ jquery.min.js:2
submit @ file-upload.js:259
(anonymous) @ file-upload.js:498
_.each._.forEach @ underscore.js:153
submitUploads @ file-upload.js:496
onNoConflicts @ file-upload.js:888
checkExistingFiles @ file-upload.js:700
add @ file-upload.js:906
_trigger @ jquery-ui.custom.js:789
(anonymous) @ jquery.fileupload.js:935
each @ jquery.min.js:2
_onAdd @ jquery.fileupload.js:928
(anonymous) @ jquery-ui.custom.js:405
(anonymous) @ jquery.fileupload.js:1105
j @ jquery.min.js:2
add @ jquery.min.js:2
always @ jquery.min.js:2
_onChange @ jquery.fileupload.js:1099
(anonymous) @ jquery-ui.custom.js:405
handlerProxy @ jquery-ui.custom.js:706
dispatch @ jquery.min.js:3
r.handle @ jquery.min.js:3
```

**Comment by PVince81** [comment]
looks like the mtime is sent as float?

**Comment by ownhopper** [comment]
it would seem so

**Comment by PVince81** [comment]
I might have an idea... Since 10.0 the web UI uses Webdav PUT and also gets the mtime from the browser when available. I guess that some browsers / OSes might provide the mtime as a decimal value instead of integer.

**Comment by PVince81** [comment]
@ownhopper @TDannhauer can you guys check the browser's network console for the "PUT" call of the upload and check the "X-OC-Mtime" header.

I checked on my local Firefox and the value is an integer: `X-OC-Mtime:"1488268924"`.

I suspect that on your systems the value is decimal.

I'll prepare a quick PR to truncate the value to see if it helps.

**Comment by TDannhauer** [comment]
My console shows "X-OC-Mtime	
1496220857.824" I'll test your PR

**Comment by PVince81** [comment]
Aha... I do wonder why you're seeing this. Does your OS, browser or FS have a better / higher resolution timer ?

In my test case I have openSUSE Leap 42.2 with BTRFS and Firefox 52.1.0 (64-bit), timestamps are always integer.

**Comment by TDannhauer** [comment]
I have stock Win10 latest release x64 with latest Firefox

**Comment by TDannhauer** [comment]
Hmm the fix does not help. any steps I missed to enable the changes file?

**Comment by PVince81** [comment]
@TDannhauer after applying the fix, please recheck the network console. Now the "OC-X-Mtime" value must not be decimal any more. If it is, then the patch was not applied properly maybe.

**Comment by TDannhauer** [comment]
ah sorry some stale cache or similar: the MTime is truncated and OC works as expected.

Thanks for the quickfix

**Comment by PVince81** [comment]
Thanks for confirming.

@phil-davis @individual-it is it possible to have an automated UI test for Web UI upload for files, and have it run on Win10 ?

**Comment by PVince81** [comment]
@ownhopper would be good if you too could confirm that the patch is working for you.

Might need Shift+F5 to refresh the page due to caching.

**Comment by phil-davis** [comment]
@PVince81 see #27615 which was making some checks for mtime being an integer. @individual-it made some tests for it with a PR back to the original repo that that came from. See https://github.com/viraj96/core/pull/1
This should be fixed in the backend so that any user of the API who sends an unusual mtime gets the file either rejected (if you give mtime "abc" or -42 or...) or the mtime massaged into shape for things like floating-point mtime (rounding to nearest int). That way it is fixed for the WebUI, and for other clients that use the API and might one day send unusual mtime formats.
@individual-it could have a look tomorrow and we can try to sort out a proposed combination of API tests and backend fix.

**Comment by ownhopper** [comment]
@PVince81 yes, patch solved the issue. Thanks for support

**Comment by phil-davis** [comment]
Reopening because this also needs a backend fix.

**Comment by individual-it** [comment]
that seems only to happen on PGSQL

`Doctrine\DBAL\Exception\DriverException: An exception occurred while executing 'UPDATE "oc_filecache" SET "mtime" = GREATEST("mtime", ?), "etag" = ? WHERE ("storage" = ?) AND ("path_hash" IN ('d41d8cd98f00b204e9800998ecf8427e', '45b963397aa40d4a0063e0d85e4fe7a1'))' with params [34.56, "592fee371da02", 4]:`

**Comment by individual-it** [comment]
This should be fixed by #28066 in the backend
